src/compute/src/compute_state.rs

```diff
@@ -1535,7 +1536,11 @@ impl IndexPeek {
             }
         }
 
-        Ok(results)
+        // Sort results. The coordinator collects sorted results from all workers and performs
+        // an N-way merge to produce the final results. This way most of the sorting happens in
+        // the workers, as compared to merging unsorted results and sorting in the coordinator thread.
+        results.sort_by(sort_by);
```

Comment:
Is this unconditional sort correct? Specifically, can we rely on the `sort_by` closure above to correctly sort the rows whether an `order_by` clause is present or not?